Sources/StreamChat/Models/MessagePinning.swift

```diff
@@ -8,14 +8,12 @@ import Foundation
 public struct MessagePinning: Equatable {
     /// The expiration date of the pinning. Infinite expiration in case it is `nil`.
     public let expirationDate: Date?
-
-    /// Private initialiser. The static initialisers should be used for better syntax and ease of use.
-    init(expirationDate: Date?) {
-        self.expirationDate = expirationDate
-    }
+    
+    /// The pinning date. Equals to current date by default.
+    public var pinnedAt = Date()
 
     /// Pins a message with infinite expiration.
-    public static let noExpiration = MessagePinning(expirationDate: nil)
+    public static var noExpiration: Self { .init(expirationDate: nil) }
```

Comment:
Why the change to `var`? 🤔

Reply:
Because of added `pinnedAt` field 🙂 When we get a pinning instance we want the timestamp to be equal to the current date, not the date computed when static `noExpiration` was accessed for the first time

Reply:
The goal of `MessagePinning` was to only set the information of the expiration date, not the `pinnedAt`. IMO we are mixing responsibilities here that should not be mixed.

Reply:
With `pinnedAt: Date` as a parameter of `pinMessage` this discussion is no longer relevant.